packages/Parallel/Application/ParallelFileProcessor.php

```diff
@@ -217,6 +222,42 @@ function ($exitCode, string $stdErr) use (&$systemErrors, $processIdentifier): v
             );
 
             $this->processPool->attachProcess($processIdentifier, $parallelProcess);
+        };
+
+        $tcpServer->on(ReactEvent::CONNECTION, function (ConnectionInterface $connection) use (&$jobs): void {
+            $inDecoder = new Decoder($connection, true, 512, 0, 4 * 1024 * 1024);
+            $outEncoder = new Encoder($connection);
+
+            $inDecoder->on(ReactEvent::DATA, function (array $data) use (&$jobs, $inDecoder, $outEncoder): void {
+                $action = $data[ReactCommand::ACTION];
+                if ($action !== Action::HELLO) {
+                    return;
+                }
+
+                $processIdentifier = $data[Option::PARALLEL_IDENTIFIER];
+                $parallelProcess = $this->processPool->getProcess($processIdentifier);
+                $parallelProcess->bindConnection($inDecoder, $outEncoder);
+
+                if ($jobs === []) {
+                    $this->processPool->quitProcess($processIdentifier);
+                    return;
+                }
+
+                $job = array_pop($jobs);
+                $parallelProcess->request([
+                    ReactCommand::ACTION => Action::MAIN,
+                    Content::FILES => $job,
+                ]);
+            });
+        });
+
+        for ($i = 0; $i < $numberOfProcesses; ++$i) {
+            // nothing else to process, stop now
+            if ($jobs === []) {
```

Comment:
is force progress bar to 100% here possible/correct?

Reply:
I tried apply progress bar to 100% here, but seems not working, it only show 97%:

```
➜  rector-src git:(df-parallel) time bin/rector process src tests rules-tests packages packages-tests --dry-run --clear-cache
 1490/1530 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░]  97%
1 file with changes
===================

1) packages/Parallel/Application/ParallelFileProcessor.php:3
```

Reply:
it seems it can be applied on `$inDecoder->on()` process on jobs empty:

```diff
if ($jobs === []) {
   $this->processPool->quitProcess($processIdentifier);
+   $this->rectorOutputStyle->progressFinish();
   return;
}
```

Reply:
The issue with it is it show different total when applied, when on progress, it shows 1530:

```
➜  rector-src git:(df-parallel) ✗ time bin/rector process src tests rules-tests packages packages-tests --dry-run --clear-cache
    0/1530 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]
```

when done, it shows 1550:

```
➜  rector-src git:(df-parallel) ✗ time bin/rector process src tests rules-tests packages packages-tests --dry-run --clear-cache
 1550/1550 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
2 files with changes
====================

1) packages/Parallel/Application/ParallelFileProcessor.php:3
```

Reply:
I see, it can actually be forced on reporting on `ConsoleOutputFormatter`:

```diff
    public function report(ProcessResult $processResult, Configuration $configuration): void
    {
+        if ($configuration->shouldShowProgressBar()) {
+            $this->rectorOutputStyle->progressFinish();
+        }
```

By this, this can show 100%:

```
➜  rector-src git:(df-parallel) ✗ time bin/rector process src tests rules-tests packages packages-tests --dry-run --clear-cache
 1530/1530 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
1 file with changes
===================

1) packages/Parallel/Application/ParallelFileProcessor.php:3
```

Reply:
@Flyingmana I create a commit to your branch https://github.com/rectorphp/rector-src/pull/3722/commits/88a755b19efbd249d9158a413a9f2db16b5129a3 👍 to handle progress bar 100% issue.